detectron2/modeling/meta_arch/retinanet.py

```diff
@@ -92,12 +93,53 @@ def __init__(self, cfg):
             cfg.MODEL.RETINANET.IOU_LABELS,
             allow_low_quality_matches=True,
         )
+        self.vis_period = cfg.VIS_PERIOD
+        self.input_format = cfg.INPUT.FORMAT
 
         pixel_mean = torch.Tensor(cfg.MODEL.PIXEL_MEAN).to(self.device).view(3, 1, 1)
         pixel_std = torch.Tensor(cfg.MODEL.PIXEL_STD).to(self.device).view(3, 1, 1)
         self.normalizer = lambda x: (x - pixel_mean) / pixel_std
         self.to(self.device)
 
+    def visualize_training(self, batched_inputs, results):
+        """
+        A function used to visualize images and intermidiate network results. It shows ground truth
+        bounding boxes on the original image and up to 20 predicted object
+        bounding boxes on the original image.
+
+        Args:
+            batched_inputs (list): a list that contains input to the model.
+            results (List[Instances]): a list of #images elements.
+        """
+        from detectron2.utils.visualizer import Visualizer
+        import numpy as np
+
+        assert len(batched_inputs) == len(
+            results
+        ), "Cannot visualize inputs and results of different sizes"
+        storage = get_event_storage()
+        max_vis_prop = 20
```

Comment:
It's named "prop" in R-CNN for "proposals". The name does not make sense here anymore. Maybe call it "max_boxes" or "max_results".